src/blog/understanding-ecmascript-part-3.md

```diff
@@ -0,0 +1,362 @@
+---
+title: 'Understanding the ECMAScript spec, part 3'
+author: '[Marja Hölttä](https://twitter.com/marjakh), speculative specification spectator'
+avatars:
+  - marja-holtta
+date: 2020-03-02 13:33:37
+tags:
+  - ECMAScript
+description: 'Tutorial on reading the ECMAScript specification'
+tweet: ''
+---
+
+... where we dive deep in the syntax!
+
+## Previous episodes
+
+In [part 1](/blog/understanding-ecmascript-part-1), we read through a simple method — `Object.prototype.hasOwnProperty` — and **abstract operations** it invokes. We familiarized ourselves with the shorthands `?` and `!` related to error handling. We encountered **language types**, **specification types**, **internal slots**, and **internal methods**.
+
+In [part 2](/blog/understanding-ecmascript-part-2), we examined a simple grammar production and how its runtime semantics are defined. In [the extra content](/blog/extra/understanding-ecmascript-part-2-extra), we also followed a long grammar production chain from `AssignmentExpression` to `MemberExpression`. In this episode, we'll go deeper in the definition of the ECMAScript language and its syntax.
+
+If you're not familiar with [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar), now it's a good time to check out the basics, since the spec uses context-free grammars to define the language.
```

Comment:
```suggestion
If you're not familiar with [context-free grammars](https://en.wikipedia.org/wiki/Context-free_grammar), now is a good time to check out the basics, since the spec uses context-free grammars to define the language.
```